### PR TITLE
Litematica integration

### PR DESCRIPTION
**Pull request by kikugie**
If litematica is installed registers additional command `/isorender selection` which uses current litematica selection coords for area render.

### CODE REVIEW

**Comment by BasiqueEvangelist** [comment]
Have you checked that this works without Litematica?

**Comment by kikugie** [comment]
Yes, it doesn't register the command if litematica is not installed @BasiqueEvangelist

**Comment by BasiqueEvangelist** [comment]
> Yes, it doesn't register the command if litematica is not installed @BasiqueEvangelist 

I know. I'm mostly asking whether you actually tried it without Litematica, as some funky class loading issues might occur if it's not present.

**Comment by kikugie** [comment]
I've tested that.

File: gradle.properties [hunk 1]
```diff
@@ -4,7 +4,8 @@ org.gradle.jvmargs=-Xmx1G
 # check these on https://fabricmc.net/develop
 minecraft_base_version=1.19
 minecraft_version=1.19.2
-yarn_mappings=1.19.2+build.11
+minecraft_version_full=1.19.2
```
Comment (gliscowo) [inline gradle.properties]: Why did you create this property? `minecraft_version` should be perfectly sufficient

File: gradle.properties [hunk 2]
```diff
@@ -4,7 +4,8 @@ org.gradle.jvmargs=-Xmx1G
...
+yarn_mappings=1.19.2+build.8
```
Comment (gliscowo) [inline gradle.properties]: I must also wonder why you downgraded the mappings

File: gradle.properties [hunk 3]
```diff
@@ -22,3 +23,7 @@ exo_version=0.1.5+1.19
 
 # https://maven.wispforest.io/io/wispforest/owo-lib/
 owo_version=0.8.2+1.19
+
+litematica_file=3929766
```
Comment (gliscowo) [inline gradle.properties]: Please link maven or files pages of all dependencies, otherwise updating them down the line is painful

**Review by gliscowo — CHANGES_REQUESTED** [review]
First of all, thanks for this PR! In general I think it looks fine, although there are some remarks about the buildscript changes I'd like to see resolved before merging
Cheers

File: gradle.properties (hunk 1 above)
Comment (kikugie) [reply gradle.properties]: Malilib would need 1.19.0 for example.

File: gradle.properties (hunk 2 above)
Comment (kikugie) [reply gradle.properties]: With litematica added and yarn build 11 gradle reload crashed. At least on my side, so I just used same version as in litematica.

File: gradle.properties (hunk 1 above)
Comment (gliscowo) [reply gradle.properties]: But then you declare 1.19.2? In any case, this should be part of the Malilib declaration and not some random property with a confusing name

File: gradle.properties (hunk 3 above)
Comment (kikugie) [reply gradle.properties]: I've added malilib maven, but there isn't for litematica. (Well there is, but it was last updated 2 years ago)

File: gradle.properties (hunk 1 above)
Comment (kikugie) [reply gradle.properties]: Well, that's a question to masa. If naming remains the same malilib 1.20 would be named 1.20.0. It doesn't make sense to add that field yet, but its going to be needed in the future.

File: gradle.properties (hunk 3 above)
Comment (gliscowo) [reply gradle.properties]: I'm aware of this, which is why I specifically mentioned the "Files" page on CurseForge. There should be a link that leads directly to a list of versions for each dependency (be that on a maven repository, CurseForge, Modrinth, whatever), searching them up every time you want to update wastes a lot of time

Comment (kikugie) [reply gradle.properties]: Pushed that

File: gradle.properties (hunk 1 above)
Comment (gliscowo) [reply gradle.properties]: Regardless, I do not want this to be a property, especially with this kind of misleading name. Please just move it into the Malilib dependency declaration

File: gradle.properties (hunk 3 above)
Comment (gliscowo) [reply gradle.properties]: Now we're 50% of the way there - when re-reading my initial comment I quite clearly see myself mentioning *all dependencies*

File: gradle.properties (hunk 1 above)
Comment (kikugie) [reply gradle.properties]: So leave it as just `fi.dy.masa.malilib:malilib-fabric-${project.minecraft_version}:${project.malilib_version}`?

Comment (gliscowo) [reply gradle.properties]: Well no, just make it the actual full name of the artifact, including the Minecraft version

File: gradle.properties (hunk 3 above)
Comment (kikugie) [reply gradle.properties]: Is that right?

Comment (gliscowo) [reply gradle.properties]: Ideally you'd link directly to where the Malilib artifacts are stored, but yes - this works well enough

File: gradle.properties (hunk 2 above)
Comment (gliscowo) [reply gradle.properties]: I'd ask you to try and make it work with latest yarn, mappings should really not affect the build process in any meaningful way

Comment (kikugie) [reply gradle.properties]: Malilib has issues with yarn build 11 
[crash-2022-09-28_21.36.31-client.txt](https://github.com/gliscowo/isometric-renders/files/9668462/crash-2022-09-28_21.36.31-client.txt)

Comment (kikugie) [reply gradle.properties]: Latest compatible is 10

Comment (kikugie) [reply gradle.properties]: @gliscowo is this good?

**Comment by kikugie** [comment]
@gliscowo latest litematica update works with yarn build 11. Its also now synced with with 0.4.1 mod version.

**Comment by gliscowo** [comment]
After looking at this PR again, I'd be happy to merge it code-quality-wise. 

I do however need to question why this is even required in the first place. I'm not a very large fan of introducing the baggage of two extra dependencies that need to be kept up to date for a feature that arguably does not bring a lot to the table. Could you explain your specific use-case? To me it would seem that you could simply recreate Litematica's selection with Isometric. Granted, this requires some extra work although to me that does not outweigh having to carry around two extra dependencies

**Comment by kikugie** [comment]
In technical Minecraft community this mod is quite common for showing contraptions. From personal experience and feedback from other people it's more convenient to make a selection with litematica, rather than built-in system or typing in coordinates. This can also be used to make render of just taken schematic or one of multiple active selections in the world.

**Comment by kikugie** [comment]
Extracted this functionality to a separate mod

**Pull request closed without merging**